localstack/services/s3/provider.py

```diff
@@ -157,6 +165,10 @@
 # max file size for S3 objects kept in memory (500 KB by default)
 S3_MAX_FILE_SIZE_BYTES = 512 * 1024
 
+# runtime cache of Lifecycle Expiration headers, as they need to be calculated everytime we fetch an object in case the
+# rules have changed
+EXPIRATION_CACHE: dict[BucketName, dict[ObjectKey, Expiration]] = defaultdict(dict)
```

Comment:
if this is a transient cache, do we maybe want to make this part of the `S3Provider` instance instead of module-global? from what i can tell, only instance methods of `S3Provider`, as well as `get_expiration_header` use the cache, so we can just make `get_expiration_header` a method of the provider. would make a for a nicer encapsulation maybe?

Reply:
Right, I didn't like the cache, thought about it in the store or in `models.py` file but didn't like either, it didn't really make sense. It being in the provider makes more sense! Thanks, will update! 👍